docs/protocol.md

```diff
@@ -1062,6 +1063,130 @@ RPC
 ```
 </details>
 
+### Catching up on past messages on a channel using paging
+
+🧭 **RPC Message** > **Query** > **Paged Catchup**
+
+By executing a paged catchup action, a client can ask the server to receive a specified number of
+past messages sorted by timestamps from oldest to newest on a specific channel. When a message ID 
+is provided, the server returns the requested number of messages on the specified channel that precede
+this message. Otherwise, the server returns the latest messages it has on that channel taking into 
+account the requested number of messages. If the specified number of messages is greater than what 
+is on the server in the current page, then the server only returns the messages it has on that page.
+
+For now, this message is to be used to retrieve chirps on the social media channel (/root/lao_id/social/user_public_key) 
+by paging when a new client joins the LAO instead of getting all the chirps at once. The user's public key is used
+to have a separate paging subchannel for each user to avoid sending irrelevant messages to other users. This paging is 
+done in an effort to reduce network traffic at catchup.
+
+This message is also to be used to retrieve chirps of a specific user profile from a subchannel "/root/lao_id/social/profile/user_public_key/specific_user_public_key"  where "user_public_key" is the same as before and "specific_user_public_key" is the public key of the user whose messages the client wants to retrieve. Paging is not deemed necessary for retrieving top chirps for now and can be done with the regular catchup message from a subchannel "/root/lao_id/social/top_chirps".
```

Comment:
I agree that paging for top chirps is not necessary, because for now top chirps are only the 3 first. But we can imagine in the future to be able to scroll down in the same paging way as in the feed, but in top chirps.

What is the top chirps channel supposed to do? Should it be specified? IMO it should send the top 3 on catchup, then send the new top 3 whenever an update is received server side (then front ends should listen to this channel).

Also just a quick note for front ends implementation: I think it would be easier for now to have separate caches for feed/top chirps/profile. Later on, some optimizations could be done (front end wise) to link the caches and avoid duplicate data (chirps) throughout those caches, but it should do without for now as there shouldn't be so much data to handle. Also, I think that we should not cache chirps of profiles (other than ours) since we will very rarely click on those profile (and even less come back to it right after).

Reply:
"IMO it should send the top 3 on catchup, then send the new top 3 whenever an update is received server side (then front ends should listen to this channel)." -> Yes that's the plan
"have separate caches for feed/top chirps/profile" -> Makes sense
"we should not cache chirps of profiles (other than ours) since we will very rarely click on those profile" -> I agree